### DISCUSSION TITLE
I have to add GTM and Firebase code in all the pages.

### DISCUSSION THREAD

**Discussion by mayank2108** [vercel/next.js#13812] [Help]
I haved added the code in my _app.js useeffect method. But due to this lighhouse is showing unused javascript code. What should be the proper placement for the initialization of these 2?

**Comment by jamesmosier** [comment]
Hi @mayank2108, this answer sounds like it would be useful: https://github.com/vercel/next.js/discussions/13682#discussioncomment-21797

A quote from that answer that may answer your question is:

> It is somewhat inevitable that some subset of JavaScript will be unused when using someone's library.
